### DISCUSSION TITLE
Provide full path to current parsing ETA template in process.env

### DISCUSSION THREAD

**Discussion by js2me** [eta-dev/eta#89] [Q&A]
Hello!   
Firstly, thanks for this powerful tool!   

I use ETA template in https://github.com/acacode/swagger-typescript-api   
And currently I have a problem with using custom `require()` function inside ETA template   

`__dirname` contains path where placed custom `required()` function   
`process.cwd()` contains path to direction of the called script which use ETA templates  

Custom raw implementation of `require()` function   
```js
require = (packageOrPath) => {
    const isPath = _.startsWith(packageOrPath, "./") || _.startsWith(packageOrPath, "../");

    if (isPath) {
      return require(path.resolve(process.cwd(), packageOrPath))
    }

    return require(packageOrPath);
  }
```    

Thanks!

**Comment by shadowtime2000** [comment]
@js2me Sorry for taking so long to respond, but what exactly is the problem with this basic require function? Can you clarify on what exactly is the question you are asking?

**Comment by js2me** [comment]
Hello @shadowtime2000   
I need a `require()` function in ETA templates which works like `require()` in NodeJS  
Or provide the ability to get path to current parsing template as NodeJS environment variable

**Comment by js2me** [comment]
Just I want to use require() in ETA templates   
like  
```ejs
Hello <%~ require("./currentUser") %>  
```

**Reply by js2me** [reply]
@shadowtime2000  or `includeFile()` works same ?

**Reply by shadowtime2000** [reply]
Can you define what you mean by `require` like load a relative Eta file? Then `includeFile` would work the same

**Reply by js2me** [reply]
I just want to call the scripts from NodeJS environment right in ETA templates.   
In example:   

**myEtaTemplate.eta**:  
```ejs
<%
const templateDataStorage = require("./templateDataStorage");

const fullName = templateDataStorage.get("fullName")
%>

<div><%~ fullName  %></div>

```


where is `"./templateDataStorage"` it is `templateDataStorage.js` script which places near with ETA template where it use.   

**templateDataStorage.js**:   

```js
const map = new Map();
map.set("fullName", "Sergey Volkov");

module.exports = map;
```
